Prepare changelog for 1.3.2 release (#563)
open-telemetry/opentelemetry-proto: CHANGELOG.md

## Unreleased

> Full list of differences found in [this compare](https://github.com/open-telemetry/opentelemetry-proto/compare/v1.3.2...main).

Reviewer: ```suggestion
The full list of changes can be found in the compare view for the respective release at <https://github.com/open-telemetry/opentelemetry-proto/releases>.
```

Otherwise markdown-link-check will block the PR since the tag won't exist yet 🙂

> ## 1.3.2 - 2024-05-28

Author: ```suggestion
## 1.3.2 - 2024-05-31
```

> ## 1.3.2 - 2024-05-31

Reviewer: ```suggestion
## 1.3.2 - 2024-06-28
```
